### DISCUSSION TITLE
Allow Widgets to be deployed to portal as a plugin - IE widgets without self hosting.

### DISCUSSION THREAD

**Discussion by david-schwartz-by** [Azure/api-management-developer-portal#2564] [Feature request]
Would like to see a pluggable widgets.  
Want to be able to code a widget, package it, and deploy it through the existing portal.  Then enable it and use it without self hosting and owning infrastructure and code. 

Please look at wordpress for examples.

**Comment by azaslonov** [comment]
Hi David, thanks for the feedback, that's kind of a plan - have a widget/container seamlessly boxed into a restricted iframe. We're planning to work on it in Jan.

Do you have a link to the WordPress example you mentioned? Maybe we could steal some ideas.

**Comment by mikebudzynski** [comment]
@david-schwartz-by On top of what @azaslonov mentioned, we also welcome contributions to the portal's codebase - we will consider incorporating every community-contributed widget in the managed portal.
